File: Solutions/2022/python/day_01.py
from setup_solutions import getPuzzleInput

year = "2022"
day = "01"
puzzleInput = getPuzzleInput(f"{year}", f"Day_{day}.txt")
puzzle = puzzleInput

def getMostCaloriesElf(puzzle):
    mostCalories = 0
    counting = 0
    
    for line in puzzle:
        line = line.replace("\n", "")
        if line == "":
            if counting > mostCalories:
                mostCalories = counting
            counting = 0
        else:
            counting += int(line)
    
    return mostCalories

def getTopThreeMostCaloriesElves(puzzle):
    elves = []
    counting = 0
    for line in puzzle:
        line = line.replace("\n", "")
        if line == "":
            elves.append(counting)
            counting = 0
        else:
            counting += int(line)
    
    elves.sort(reverse=True)
    return sum(elves[:3])


# Script runs here
print(f"Advent of Code - {year}: Day {day}")
print("Part-One Answer:", getMostCaloriesElf(puzzle))
print("Part-Two Answer:", getTopThreeMostCaloriesElves(puzzle))

File: Solutions/javascript/2022/day_01.js
const { getPuzzleInput } = require("../SetupSolutions/SetupSolutions.js");

const year = "2022";
const day = "01";
const puzzleInput = getPuzzleInput(year, `Day_${day}.txt`);
const puzzle = puzzleInput;

const getMostCaloriesElf = (puzzle) => {
    let mostCalories = 0;
    let count = 0;
    for (calories of puzzle) {
        if (calories === "") {
            if (count >= mostCalories) mostCalories = count;
            count = 0;
        }
        else {
            count += parseInt(calories);
        }
    }

    return mostCalories;
};

const getTopThreeElves = (puzzle) => {
    let elves = [];
    let count = 0;
    for (calories of puzzle) {
        if (calories === "") {
            elves.push(count);
            count = 0;
        }
        else {
            count += parseInt(calories);
        }
    }

    return elves.sort((a,b) => b - a)
                .slice(0, 3)
                .reduce((prev, curr) => prev + curr, 0)
};

// Script runs here
console.log(`Advent of Code - ${year}: Day ${day}`);
console.log("Part-One Answer:", getMostCaloriesElf(puzzle));
console.log("Part-Two Answer:", getTopThreeElves(puzzle));